builder/comp-builder.nix

```diff
@@ -240,9 +241,11 @@ stdenv.mkDerivation ({
   # Phases
   preInstallPhases = lib.optional doHaddock' "haddockPhase";
 
-  prePatch = lib.optionalString (cabalFile != null) ''
-    cat ${cabalFile} > ${package.identifier.name}.cabal
-  '';
+  prePatch = if (cabalFile != null)
+     then ''cat ${cabalFile} > ${package.identifier.name}.cabal''
+     else if (cabal-generator == "hpack")
```

Comment:
```suggestion
     else lib.optionalString (cabal-generator == "hpack") ''
       ${hsPkgs.hpack.components.exes.hpack}/bin/hpack
     '';
```

Reply:
Not sure if we need parens here.

Reply:
@angerman Why is this taking hpack from hsPkgs? It ends up differing from callCabalProjectToNix how uses it and appears to be the root of why my use of that is failing when running the resultant derivation.